NetKet/Hamiltonian/MatrixWrapper/matrix_wrapper.hpp

```diff
@@ -0,0 +1,40 @@
+#ifndef MATRIX_WRAPPER_HPP
+#define MATRIX_WRAPPER_HPP
+
+#include <memory>
+
+#include "Graph/graph.hpp"
+#include "Hamiltonian/hamiltonian.hpp"
+#include "Utils/json_helper.hpp"
+
+#include "dense_matrix_wrapper.hpp"
+#include "direct_matrix_wrapper.hpp"
+#include "sparse_matrix_wrapper.hpp"
+
+namespace netket {
+
+template<class Wrapped>
+std::unique_ptr<AbstractMatrixWrapper<Wrapped>>
+ConstructMatrixWrapper(const json &pars, const Wrapped& wrapped)
+{
+    using WrapperPtr = std::unique_ptr<AbstractMatrixWrapper<Wrapped>>;
+
+    std::string wrapper_name = FieldOrDefaultVal<json, std::string>(pars, "MatrixWrapper", "Sparse");
+    if(wrapper_name == "Sparse")
+    {
+        return WrapperPtr(new SparseMatrixWrapper<Wrapped>(wrapped));
+    }
+    else if(wrapper_name == "Dense")
+    {
+        return WrapperPtr(new SparseMatrixWrapper<Wrapped>(wrapped));
+    }
+    else
+    {
+        std::cout << "Unknown MatrixWrapper: " << wrapper_name << std::endl;
+        std::abort();
```

Comment:
Why'd you call abort here? Throwing an exception should do the job.

Reply:
In principle I very much agree. Since calling `abort` on invalid input is currently done all over the code, however, I planned to do just the same here for now and discuss improving the error handling by using exceptions in another pull request.

Reply:
Indeed the `abort` solution I adopted is very bad, and an ad interim solution waiting for a better system to intercept errors

Reply:
@gcarleo I would suggest changing all places were `abort` is called to throw a suitable exception and then handle it in the main method by printing the error message and exiting. That should be sufficient for invalid input. I could prepare a pull request with that in the next few days, if you agree.

Reply:
Sounds good. Standard arguments against using exceptions (predictable performance etc.) do not apply to this projects, so I would suggest to stick to them.